lib/vec.rs

```diff
@@ -0,0 +1,20 @@
+use crate::closure::Closure;
+use crate::ddlog_std;
+
+pub fn vec_sort_by<A, B: Ord>(v: &mut ddlog_std::Vec<A>, f: &Box<dyn Closure<*const A, B>>) {
+    v.x.sort_unstable_by_key(|x| f.call(x))
```

Comment:
if there is a stable sort, why not expose that one too?

Reply:
We can also have `vec_sort_by_stable`.  Stable sort is less efficient though.